documentation/design_docs/L2/tsop752.md

```diff
@@ -0,0 +1,131 @@
+# Tsop752 Infrared Receiver Design Document
+
+- [Tsop752 Infrared Receiver Design Document](#tsop752-infrared-receiver-design-document)
+- [Location](#location)
+- [Type](#type)
+- [Background](#background)
+- [Overview](#overview)
+- [Detailed Design](#detailed-design)
+  - [API](#api)
+    - [Initialization](#initialization)
+    - [Capturing The De-modulated IR signal](#capturing-the-de-modulated-ir-signal)
+- [Caveats](#caveats)
+- [Future Advancements](#future-advancements)
+- [Testing Plan](#testing-plan)
+  - [Unit Testing Scheme](#unit-testing-scheme)
+  - [Demonstration Project](#demonstration-project)
+
+# Location
+`L2 HAL`
+
+# Type
+Implementation
+
+# Background
+The Tsop752 is an infrared (IR) receiver module for IR remote control systems.
+The device detects modulated IR signals and outputs a de-modulated signal to be
+processed by a micro-controller.
+
+The device data-sheet can be found
+[here](https://www.vishay.com/docs/82494/tsop752.pdf).
+
+# Overview
+The Tsop752 outputs IR data frames which consist of a de-modulated pulse train.
+The `Capture` module is used to capture incoming pulses of each frame from the
+device. When a rising and/or falling edge of a signal is detected, a timestamp
+in milliseconds is be recorded. The logic level of data in the received frame
+is determined by examining the period of each pulse.
+
+In addition to the capture module, the `Timer` module will be utilized to
+determine the end of transmission frame when no consecutive pulse is received
+within a timeout period after the last received pulse.
+
+For consumer IR applications, such as IR remote control systems, two of
+commonly used IR transmission protocol include RC-5 and NEC. An example of a
+typical IR transmission frame conforming to the NEC protocol can be found
+[here](https://techdocs.altium.com/sites/default/files/wiki_attachments/296329/NECMessageFrame.png).
+
+# Detailed Design
+## API
+```C++
+namespace sjsu
+{
+class Tsop752 final : public InfraredReceiver
+{
+ public:
+  constexpr explicit Tsop752(sjsu::Capture & capture,
+                             sjsu::Timer & timer);
```

Comment:
I don't understand why you need both a timer and a capture peripheral.

Reply:
The timer helps automatically detect the end of a transmission after not receiving a pulse after a duration or should the user just specify how many pulses to be expected per frame?